src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1396,7 +1396,15 @@ private CSharpSyntaxNode GetBindingRoot(CSharpSyntaxNode node)
                 {
                     case SyntaxKind.ThisConstructorInitializer:
                     case SyntaxKind.BaseConstructorInitializer:
+                        return current;
                     case SyntaxKind.ArrowExpressionClause:
+                        // If this is an arrow expression on a local function statement, then our bindable root is actually our parent syntax as it's
+                        // a statement in a function. If this is returned directly in IOperation, we'll end up with a separate tree.
+                        if (current.ParentOrStructuredTriviaParent != null && current.ParentOrStructuredTriviaParent.Kind() == SyntaxKind.LocalFunctionStatement)
```

Comment:
>if (current.ParentOrStructuredTriviaParent != null && current.ParentOrStructuredTriviaParent.Kind() == SyntaxKind.LocalFunctionStatement) [](start = 24, length = 137)

It is unfortunate that we have to do this. In SemanticModel we want to bind as little as possible, if we bubble up to the entire local function, we will be forced to bind entire block body, if it is present, and that block body could be quite big. From what I see things should work properly with the original implementation. It looks like ```MethodBodySemanticModel.Bind``` should be able to bind ArrowExpressionClause on its own and the problem is that that node is not getting attached to the BoundLocalFunctionStatement when we bind entire local function statement. I suspect that is happening because either incremental binder is not used to bind local function statement,  or the node is getting mangled by ```runAnalysis``` transformation in BindLocalFunctionStatement and, therefore, is not matched to what we already have in the cache. Also, the way block body is bound in BindLocalFunctionStatement prevents its reuse (see how Binder.BindMethodBody binds block so that it can be reused by SemanticModel). Anyway, all this looks like a "preexisting condition", we don't have to address all this in this PR, at least we should open an issue to follow up (with all this information).

Reply:
Filed [https://github.com/dotnet/roslyn/issues/25815](https://github.com/dotnet/roslyn/issues/25815) for followup.

---
In reply to: [177942610](https://github.com/dotnet/roslyn/pull/25782#discussion_r177942610) [](ancestors = 177942610)